lib/routes/lnurl/payment/lnurl_payment_handler.dart

```diff
@@ -0,0 +1,113 @@
+      useRootNavigator: false,
+      context: context,
+      barrierDismissible: false,
+      builder: (_) => LNURLPaymentDialog(requestData: requestData),
+    );
+  } else {
+    paymentInfo = await Navigator.of(context).push<LNURLPaymentInfo>(
+      FadeInRoute(
+        builder: (_) => LNURLPaymentPage(requestData: requestData),
+      ),
+    );
+  }
+  if (paymentInfo == null) {
+    return Future.value();
+  }
+  // Artificial wait for UX purposes
+  await Future.delayed(const Duration(milliseconds: 800));
+  // Show Processing Payment Dialog
+  // ignore: use_build_context_synchronously
+  return await showDialog(
+    useRootNavigator: false,
+    context: context,
+    barrierDismissible: false,
+    builder: (_) => ProcessingPaymentDialog(
+      isLnurlPayment: true,
+      firstPaymentItemKey: firstPaymentItemKey,
+      paymentFunc: () => context.read<AccountBloc>().lnurlPay(
+            amount: paymentInfo!.amount,
+            comment: paymentInfo.comment,
+            reqData: requestData,
+          ),
+    ),
+  ).then((result) {
+    if (result is LnUrlPayResult) {
+      if (result is LnUrlPayResult_EndpointSuccess) {
+        _log.v("LNURL payment success, action: ${result.data}");
+        return LNURLPageResult(
+          protocol: LnUrlProtocol.Pay,
+          successAction: result.data,
+        );
+      } else if (result is LnUrlPayResult_EndpointError) {
+        _log.v("LNURL payment failed: ${result.data.reason}");
+        return LNURLPageResult(
+          protocol: LnUrlProtocol.Pay,
+          error: result.data.reason,
+        );
+      }
+    }
+    _log.w("Error sending LNURL payment", ex: result);
+    throw LNURLPageResult(error: result).errorMessage;
+  });
+}
+
+void handleLNURLPaymentPageResult(BuildContext context, LNURLPageResult result) {
+  if (result.successAction != null) {
+    _handleSuccessAction(context, result.successAction!);
+  } else if (result.hasError) {
+    _log.v("Handle LNURL payment page result with error '${result.error}'");
+    throw Exception(result.errorMessage);
+  }
+}
+
+Future _handleSuccessAction(BuildContext context, SuccessActionProcessed successAction) {
+  String message = '';
+  String? url;
+  if (successAction is SuccessActionProcessed_Message) {
+    message = successAction.data.message;
+    _log.v("Handle LNURL payment page result with message action '$message'");
+  } else if (successAction is SuccessActionProcessed_Url) {
+    message = successAction.data.description;
+    url = successAction.data.url;
+    _log.v("Handle LNURL payment page result with url action '$message', '$url'");
+  } else if (successAction is SuccessActionProcessed_Aes) {
+    message = "${successAction.data.description} ${successAction.data.plaintext}";
+    _log.v("Handle LNURL payment page result with aes action '$message'");
+  }
+  // Artificial delay for UX purposes
+  return Future.delayed(const Duration(seconds: 1)).then(
```

Comment:
To give a breathing room between dialogs. Otherwise this dialog opens instantly once closing animation of previous dialog starts, we don't want to scare older generations with [this](https://youtu.be/9sycZ4GnUA4) phantom sound😅

Reply:
I did a comparison and decided to remove these delays.
Will send you the comparison video.